deploy/gcp/main.tf

```diff
@@ -0,0 +1,290 @@
+variable "GCP_CREDENTIALS_PATH" {}
+variable "GCP_REGION" {}
+variable "GCP_PROJECT" {}
+
+provider "google" {
+  credentials = "${file("${var.GCP_CREDENTIALS_PATH}")}"
+  region = "${var.GCP_REGION}"
+  project = "${var.GCP_PROJECT}"
+}
+
+// required for taints on node pools
+provider "google-beta" {
+  credentials = "${file("${var.GCP_CREDENTIALS_PATH}")}"
+  region = "${var.GCP_REGION}"
+  project = "${var.GCP_PROJECT}"
+}
+
+locals {
+  credential_path = "${path.module}/credentials"
+  kubeconfig                    = "${local.credential_path}/kubeconfig_${var.cluster_name}"
+  tidb_cluster_values_path = "${path.module}/rendered/tidb-cluster-values.yaml"
+}
+
+resource "null_resource" "prepare-dir" {
+  provisioner "local-exec" {
+    command = "mkdir -p ${local.credential_path}"
+  }
+}
+
+resource "google_compute_network" "vpc_network" {
+  name = "vpc-network"
+  auto_create_subnetworks = false
+  project = "${var.GCP_PROJECT}"
+}
+
+resource "google_compute_subnetwork" "private_subnet" {
+  ip_cidr_range = "172.31.252.0/22"
+  name = "private-subnet"
+  network = "${google_compute_network.vpc_network.self_link}"
+  project = "${var.GCP_PROJECT}"
+  secondary_ip_range {
+    ip_cidr_range = "172.30.0.0/16"
+    range_name = "pods-${var.GCP_REGION}"
+  }
+  secondary_ip_range {
+    ip_cidr_range = "172.31.224.0/20"
+    range_name = "services-${var.GCP_REGION}"
+  }
+}
+
+resource "google_compute_subnetwork" "public_subnet" {
+  ip_cidr_range = "172.29.252.0/22"
+  name = "public-subnet"
+  network = "${google_compute_network.vpc_network.self_link}"
+  project = "${var.GCP_PROJECT}"
+}
+
+resource "google_container_cluster" "cluster" {
+  name = "${var.cluster_name}"
+  network = "${google_compute_network.vpc_network.self_link}"
+  subnetwork = "${google_compute_subnetwork.private_subnet.self_link}"
+  location = "${var.GCP_REGION}"
+  project = "${var.GCP_PROJECT}"
+
+  master_auth {
+    username = ""
+    password = ""
```

Comment:
What does the empty password mean? Disable password login or allow login with empty password?

Reply:
It's for disabling, otherwise I believe GKE generates a password with username `admin`